Demo/Demo/ViewModels/PayPalViewModel.swift

```diff
@@ -109,26 +123,31 @@ extension PayPalViewModel: PayPalDelegate {
         shippingChange: ShippingChange,
         shippingChangeAction: ShippingChangeAction
     ) {
-        switch shippingChange.type {
-        case .shippingAddress:
-            // If user selected new address, we generate new shipping methods
-            let availableShippingMethods = OrderRequestHelpers.getShippingMethods(baseValue: Int.random(in: 0..<6))
-
-            // If shipping methods are available, then patch order with the new shipping methods and new amount
-            patchAmountAndShippingOptions(
-                shippingMethods: availableShippingMethods,
-                action: shippingChangeAction
-            )
-
-        case .shippingMethod:
-            // If user selected new method, we patch the selected shipping method + amount
-            patchAmountAndShippingOptions(
-                shippingMethods: shippingChange.shippingMethods,
-                action: shippingChangeAction
-            )
-
-        @unknown default:
-            break
+        // A bug in NXO calls callback everytime. We have to store shipping preference for cases with no shipping
```

Comment:
Interesting - so the `getFromFile` option is basically a workaround for this NXO bug?

Reply:
@scannillo yep. The callback should only be called when an order has `GET_FROM_FILE` as its `shippingPreference`. But now its being called every time, so i have to filter it here for now. If i try to patch an order that has `NO_SHIPPING` or `SET_PROVIDED_ADDRESS`, i get an error while patching

Reply:
Got it - this filter makes senes as a workaround. My only comment is "get from file" as a name is a bit confusing to me.